src/pages/TrueSight/hooks/useGetTrendingSoonData.ts

```diff
@@ -47,69 +47,54 @@ export interface TrueSightTokenResponse {
-  const [error, setError] = useState<Error>()
-  const [data, setData] = useState<TrueSightTokenResponse>()
+  const { data, error } = useSWR(
+    `${process.env.REACT_APP_TRUESIGHT_API}/api/v1/trending-soon?timeframe=${
+      filter.timeframe === TrueSightTimeframe.ONE_DAY ? '24h' : '7d'
+    }&page_number=0&page_size=${maxItems}&search_token_id=${
+      filter.selectedTokenData?.token_id ?? ''
+    }&search_token_tag=${filter.selectedTag ?? ''}`,
+    async (url: string) => {
+      const response = await fetch(url)
+      if (response.ok) {
+        const json = await response.json()
+        let result: TrueSightTokenResponse = json.data
 
-  useEffect(() => {
-    const fetchData = async () => {
-      try {
-        const timeframe = filter.timeframe === TrueSightTimeframe.ONE_DAY ? '24h' : '7d'
-        const url = `${
-          process.env.REACT_APP_TRUESIGHT_API
-        }/api/v1/trending-soon?timeframe=${timeframe}&page_number=0&page_size=${maxItems}&search_token_id=${
-          filter.selectedTokenData?.token_id ?? ''
-        }&search_token_tag=${filter.selectedTag ?? ''}`
-        setError(undefined)
-        setIsLoading(true)
-        const response = await fetch(url)
-        if (response.ok) {
-          const json = await response.json()
-          let result: TrueSightTokenResponse = json.data
-
-          // Sort platforms
-          result.tokens = result.tokens.map(token => {
-            const priorityNetworks = Object.keys(TRENDING_SOON_SUPPORTED_NETWORKS)
-            const platforms = new Map<string, string>()
-            for (let i = 0; i < priorityNetworks.length; i++) {
-              const network = priorityNetworks[i]
-              const address = (token.platforms as unknown as { [p: string]: string })[network]
-              if (address) {
-                platforms.set(network, address)
-              }
-            }
-            return {
-              ...token,
-              platforms,
-            }
-          })
-
-          // Filter network in frontend
-          if (filter.selectedNetwork) {
-            const selectedNetworkKey = Object.keys(TRENDING_SOON_SUPPORTED_NETWORKS).find(
-              (key: string) => TRENDING_SOON_SUPPORTED_NETWORKS[key] === filter.selectedNetwork,
-            )
-            const filteredTokens = result.tokens.filter(tokenData =>
-              tokenData.present_on_chains.includes(selectedNetworkKey as string),
-            )
-            result = {
-              total_number_tokens: filteredTokens.length,
-              tokens: filteredTokens,
+        // Sort platforms
+        result.tokens = result.tokens.map(token => {
+          const priorityNetworks = Object.keys(TRENDING_SOON_SUPPORTED_NETWORKS)
+          const platforms = new Map<string, string>()
+          for (let i = 0; i < priorityNetworks.length; i++) {
+            const network = priorityNetworks[i]
+            const address = (token.platforms as unknown as { [p: string]: string })[network]
```

Comment:
it's Map, you can get it by:
```
  const address = token.platforms.get(network)
```

Reply:
No, it's a normal JS object. It's transforming to Map.
```
// Sort platforms
          result.tokens = result.tokens.map(token => {
            const priorityNetworks = Object.keys(TRENDING_SOON_SUPPORTED_NETWORKS)
            const platforms = new Map<string, string>()
            for (let i = 0; i < priorityNetworks.length; i++) {
              const network = priorityNetworks[i]
              const address = (token.platforms as unknown as { [p: string]: string })[network]
              if (address) {
                platforms.set(network, address)
              }
            }
            return {
              ...token,
              platforms,
            }
          })
```